src/godocToMarkdown.ts

```diff
@@ -0,0 +1,372 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------*/
+
+import * as XRegExp from 'xregexp';
+
+function markdownEscape(text: string): string {
+	return text.replace(/([\\`*{}[\]()#+-.!_>~|\n"$%&',\/:;<=?@^])/g, '\\$1');
```

Comment:
@segevfiner where is this regex from ? I couldn't see it in the go impl

Reply:
It isn't in the Go impl. The Go impl does plain text and HTML. This one does Markdown. In order to get Markdown that renders the same as the HTML we need to escape anything that would be interpreted as markup by Markdown that isn't part of the minimal processing that godoc does.

This is basically a list of characters that can have special meaning in Markdown and need to be backslash escaped.

I based this on this: https://www.npmjs.com/package/markdown-escapes

Reply:
@segevfiner thanks for the quick reply. ill put the reference next to it